tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2966,84 +2966,152 @@ llvm::Optional<Value> convertResizeOp(PatternRewriter& rewriter, Operation* op,
   size_t output_height = output_shape[1];
   size_t output_width = output_shape[2];
 
-  double fp_stride_y =
-      static_cast<double>(input_height) / static_cast<double>(output_height);
-  double fp_stride_x =
-      static_cast<double>(input_width) / static_cast<double>(output_width);
-  if (align_corners && output_height > 1) {
-    fp_stride_y = static_cast<double>(input_height - 1) /
-                  static_cast<double>(output_height - 1);
-  }
-  if (align_corners && output_width > 1) {
-    fp_stride_x = static_cast<double>(input_width - 1) /
-                  static_cast<double>(output_width - 1);
-  }
-
-  double fp_offset_y, fp_offset_x;
-  if (half_pixel_centers) {
-    fp_offset_y = fp_stride_y * 0.5f - 0.5f;
-    fp_offset_x = fp_stride_x * 0.5f - 0.5f;
-  } else {
-    fp_offset_y = 0.0f;
-    fp_offset_x = 0.0f;
+  // a. For power of two upscale [OH, OW] = (1 << k) * [IH, IW],
+  // sampling range approximately (-0.5, -0.5) to (IH - 0.5, IW - 0.5), set:
+  // scale_y_n = 2 << k, scale_y_d = 2, offset_y = -(1 << k) + 1, border_y = (1
```

Comment:
Nit: could you break this over multiple lines in way that keeps the formulas together? Perhaps one per line would avoid breaking inside a formula

Reply:
Your suggestion is nicer. Done.